### PR TITLE
Add support for Swift Package Manager

### PR DESCRIPTION
**Pull request by natestedman**
This adds support for the [Swift Package Manager](https://github.com/apple/swift-package-manager). Due to https://github.com/apple/swift-package-manager/pull/16, I had to rename the `ResultTests` directory to `Tests`. Building and testing in Xcode and with `carthage` still works correctly.

You can test it with this `Package.swift` file:

```swift
import PackageDescription

let package = Package(
    name: "Testing",
    dependencies: [
        .Package(url: "https://github.com/natestedman/Result.git", majorVersion: 1, minor: 1)
    ]
)
```

```
$ /Library/Developer/Toolchains/swift-2.2-SNAPSHOT-2015-12-01-a.xctoolchain/usr/bin/swift build
Compiling Swift Module 'Result' (2 sources)
Linking Library:  .build/debug/Result.a
```

I had to tag an additional release on my fork to try it out, as I don't believe that the Swift Package Manager currently supports arbitrary branches or commit SHAs like `carthage` does.

### CODE REVIEW

**Comment by NachoSoto** [comment]
:heart_eyes::heart_eyes::heart_eyes::heart_eyes:

File: Result.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -191,7 +191,8 @@
 				D454806E1A9572F5009D7229 /* ResultTests.swift */,
 				D454806C1A9572F5009D7229 /* Supporting Files */,
 			);
-			path = ResultTests;
+			name = ResultTests;
```
Comment (robrix) [inline Result.xcodeproj/project.pbxproj]: We can rename the group too if it makes things easier. I’ve no preference.

Comment (robrix) [inline Result.xcodeproj/project.pbxproj]: I’m a little surprised we don’t have to do the same thing with the sources as well.

**Comment by robrix** [comment]
Thank you for the PR!

**Comment by swizzlr** [comment]
Could this get merged?

**Comment by robrix** [comment]
Needs a merge from master, to begin with.

**Comment by natestedman** [comment]
Rebased.

**Comment by robrix** [comment]
Thanks!

**Pull request merged**